services/minespace-web/src/components/Forms/projects/majorMineApplication/MajorMineApplicationForm.js

```diff
@@ -209,50 +234,70 @@ export class MajorMineApplicationForm extends Component {
-              );
-            }}
-            projectGuid={this.props.project?.project_guid}
-            labelIdle={
-              '<strong>Drag & Drop your files or <span class="filepond--label-action">Browse</span></strong><div>(Accepted filetypes: .kmx .doc .docx .xlsx .pdf)</div>'
-            }
-            allowMultiple
-            acceptedFileTypesMap={this.acceptedFileTypesMap}
-            component={MajorMineApplicationFileUpload}
-            uploadType="spatial_document"
-          />
-          {spatialDocuments?.length > 0 && (
-            <DocumentTable
-              documents={spatialDocuments}
-              documentColumns={documentColumns}
-              documentParent="Major Mine Application"
-              canArchiveDocuments={true}
-              onArchivedDocuments={() => this.props.refreshData()}
-              enableBulkActions={true}
-            />
+          <br />
+          {this.props.isFeatureEnabled(Feature.MAJOR_PROJECT_ALL_DOCUMENTS) ? (
+            <>
+              <Button
+                onClick={this.openSpatialDocumentModal}
+                type="primary"
+                className="block-button"
+                style={{ marginTop: 12 }}
+              >
+                Upload Spatial Data
+              </Button>
+              {this.props.spatial_documents?.length > 0 && (
+                <SpatialDocumentTable documents={this.props.spatial_documents} />
+              )}
+            </>
+          ) : (
+            <>
+              <Field
+                id={MAJOR_MINES_APPLICATION_DOCUMENT_TYPE.SPATIAL}
+                name={MAJOR_MINES_APPLICATION_DOCUMENT_TYPE.SPATIAL}
+                label="Upload spatial components"
+                onFileLoad={(documentName, document_manager_guid) => {
+                  this.onFileLoad(
+                    documentName,
+                    document_manager_guid,
+                    MAJOR_MINES_APPLICATION_DOCUMENT_TYPE_CODE.SPATIAL,
+                    MAJOR_MINES_APPLICATION_DOCUMENT_TYPE.SPATIAL
+                  );
+                }}
+                onRemoveFile={(err, fileItem) => {
+                  this.onRemoveFile(
+                    err,
+                    fileItem,
+                    MAJOR_MINES_APPLICATION_DOCUMENT_TYPE.SPATIAL,
+                    this.props.spatial_documents
+                  );
+                }}
+                projectGuid={this.props.project?.project_guid}
+                labelIdle={
+                  '<strong>Drag & Drop your files or <span class="filepond--label-action">Browse</span></strong><div>(Accepted filetypes: .kmx .doc .docx .xlsx .pdf)</div>'
+                }
+                allowMultiple
+                acceptedFileTypesMap={this.acceptedFileTypesMap}
+                component={MajorMineApplicationFileUpload}
+                uploadType="spatial_document"
+              />
+              {spatialDocuments?.length > 0 && (
+                <DocumentTable
```

Comment:
Please see the DocumentUpload or ProjectDocumentsTab files!
For spatial documents, we should be using the components there. The SpatialDocumentsTable, and not the regular uploader, it needs to be the spatial documents uploader and not the regular one or the MajorMineApplicationFileUpload to do the validation properly.

Would also love to see this file converted to TS if possible, it generates TS errors in my console all the time...